esx_service/vmdk_ops.py

```diff
@@ -739,16 +744,34 @@ def get_full_vol_name(vmdk_name, datastore):
 
 
 # TBD - move to vmdk_utils
-def get_datastore_name(config_path):
-    """Returns datastore NAME in config_path (not url-name which may be used in path)"""
-    # path is always /vmfs/volumes/<datastore>/... , so extract datastore:
-    config_ds_name = config_path.split("/")[3]
-    ds_name = [x[0] for x in vmdk_utils.get_datastores() \
-                    if x[0] == config_ds_name or x[1] == config_ds_name ]
-    if len(ds_name) != 1:
-        logging.error("get_datastore_name: found more than one match: %s" % ds_name)
-    logging.debug("get_datastore_name: path=%s name=%s" % (config_ds_name, ds_name))
-    return ds_name[0]
+def get_datastore_url_from_config_path(config_path):
```

Comment:
can this be moved to vmdk_utils as part of this PR (seems the whole method is refactored)?